src/js/libgeo/GeoOps.js

```diff
@@ -584,6 +584,43 @@ geoOps.OtherPointOnCircle.updatePosition = function(el) {
     el.homog = General.withUsage(pos, "Point");
 };
 
+
+// start and end assumed to be normalized at once!
+geoOps._helper.projectPointToSegmentCR = function(seg, pos, infseg) {
+    var line = seg.homog;
+    var tt = List.turnIntoCSList([line.value[0], line.value[1], CSNumber.zero]);
+    var farpoint = List.sub(seg.startpos, seg.endpos);
+    var cr = List.crossratio3(
+        farpoint, seg.startpos, seg.endpos, pos, tt);
+
+    //handle case if we have infinite segment
+    if (infseg) {
```

Comment:
This is intended to handle the case when a PointOnArc passes through infinity while the defining points of the arc do not. I think this is a different case, i'm note sure if there is easier way.
![a1](https://cloud.githubusercontent.com/assets/14185442/18581733/252c7ed8-7c02-11e6-8c4a-89b30046be25.png)
![a2](https://cloud.githubusercontent.com/assets/14185442/18581732/25279c42-7c02-11e6-9402-df2b42d0dc92.png)